src/gui/spatialArrow.js

```diff
@@ -0,0 +1,224 @@
+                
+                screenX = screenXY.x;
+                screenY = screenXY.y;
+
+                desX = remap(screenX, 0, screenW, -screenW/2, screenW/2);
+                desY = remap(screenY, 0, screenH, -screenH/2, screenH/2);
+                
+                angle = Math.atan2(desY, desX);
+                angle += Math.PI / 2;
+
+                k = (screenY - screenH / 2) / (screenX - screenW / 2);
+
+                if (k < 0) {
+                    if (Math.abs(k) < screenRatio) {
+                        if (screenX < screenW / 2) {
+                            // left side bottom half
+                            finalPosX = - screenW / 2;
+                            finalPosY = finalPosX * k;
+                        } else {
+                            // right side top half
+                            finalPosX = screenW / 2;
+                            finalPosY = finalPosX * k;
+                        }
+                    } else {
+                        if (screenX < screenW / 2) {
+                            // bottom side left half
+                            finalPosY = screenH / 2;
+                            finalPosX = finalPosY / k;
+                        } else {
+                            // top side right half
+                            finalPosY = - screenH / 2;
+                            finalPosX = finalPosY / k;
+                        }
+                    }
+                } else {
+                    if (Math.abs(k) < screenRatio) {
+                        if (screenX < screenW / 2) {
+                            // left side top half
+                            finalPosX = - screenW / 2;
+                            finalPosY = finalPosX * k;
+                        } else {
+                            // right side bottom half
+                            finalPosX = screenW / 2;
+                            finalPosY = finalPosX * k;
+                        }
+                    } else {
+                        if (screenX < screenW / 2) {
+                            // top side left half
+                            finalPosY = - screenH / 2;
+                            finalPosX = finalPosY / k;
+                        } else {
+                            // bottom side right half
+                            finalPosY = screenH / 2;
+                            finalPosX = finalPosY / k;
+                        }
+                    }
+                }
+
+                finalPosX *= screenBorderFactor;
+                finalPosY *= screenBorderFactor;
+
+                drawArrow(finalPosX, finalPosY, angle, 5, indicator.avatarColor, indicator.avatarColorLighter);
+            }
+        })
+    }
+    
+    function drawIndicatorArrows() {
+        searchForIndicators();
+        determineObjectScreenPosition();
+        // draw a single test arrow at the screen center
+        
```

Comment:
It looks like the function `determineObjectScreenPosition` is also performing the draw - maybe it can be named more clearly like `drawArrowAtScreenPositionOfEachIndicator` (or something a bit shorter that still gets the point across)

Also, is the comment `// draw a single test arrow at the screen center` referring to the line above it, or is it referring to code that was removed? (in which case it should be removed)